File: Qt/console/console_board.cpp
#include "console_board.h"
#include "abstract_player.h"
#include "computer_player.h"

ConsoleBoard::ConsoleBoard(Game *game)
{
//  cout << "Constructing ConsoleBoard" << endl;
  this->game = game;
}

ConsoleBoard::~ConsoleBoard()
{
//  cout << "Destroying ConsoleBoard" << endl;
}

void ConsoleBoard::print()
{
  vector<Card *> & cards = game->getPlayer1()->getCards();
  AbstractPlayer *player = game->getPlayer1();
  ComputerPlayer *compPlayer = dynamic_cast<ComputerPlayer *>(player);

  cout << *player << "'s cards are: ";
  for (int i=0; i<cards.size(); i++)
  {
    Card *card = cards.at(i);
    if (compPlayer)
    {
      cout << "XX ";
    }
    else
    {
      cout << *card << " ";
    }
  }
  cout << endl;
}

void ConsoleBoard::fireCardMove(Card *card)
{
}

void ConsoleBoard::fireTakeCards(bool isOne)
{
}

void ConsoleBoard::fireResetMoves()
{
}

File: Android/src/com/bu6ido/jsantase/boards/ConsoleBoard.java
/**
 * 
 */
package com.bu6ido.jsantase.boards;

import java.util.List;

import com.bu6ido.jsantase.Game;
import com.bu6ido.jsantase.common.PlayCard;

/**
 * @author bu6ido
 *
 */
public class ConsoleBoard implements IBoard {

	private Game game;
	
	public ConsoleBoard(Game game)
	{
		this.game = game;
	}
	
	@Override
	public void print() 
	{
		List<PlayCard> cards = game.isOne()? 
			game.getPlayer1().getCards() : game.getPlayer2().getCards();
		System.out.println("Your cards are: ");
		for (int i=0; i<cards.size(); i++)
		{
			PlayCard card = cards.get(i);
			System.out.print(card + " ");
		}
		System.out.println();
	}

	@Override
	public void fireCardMove(PlayCard card) 
	{
		
	}
	
	@Override
	public void fireTakeCards(boolean isOne) 
	{
		
	}
}
